Makefile

```diff
@@ -0,0 +1,23 @@
+.PHONY: all generate cargo diff
+
+all: diff
+
+generate:
+	@echo "Generating files in tools.go"
+	@go generate -v ./cmd/gen-docs
+	@go generate -v ./internal/config/config.go
+	@go generate -v ./pkg/contracts/main.go
+	@go generate -v ./pkg/readerclient/main.go
```

Comment:
We should use `go generate ./...` instead of explicitly declaring each file here.